### ISSUE TITLE
multiclass

### ISSUE THREAD

**Issue by HuangQinJian** [qubvel/segmentation_models.pytorch#29]
If I have four class,
```
class DiceLoss(nn.Module):
    __name__ = 'dice_loss'

    def __init__(self, eps=1e-7, activation='softmax2d'):
        super().__init__()
        self.activation = activation
        self.eps = eps

    def forward(self, y_pr, y_gt):
        return 1 - F.f_score(y_pr, y_gt, beta=1., eps=self.eps, threshold=None, activation=self.activation)
```

activation='softmax2d'  should be set 'softmax2d' by myself?

**Comment by HuangQinJian** [comment]
Also, I must modify the train.py code 

```
with tqdm(dataloader, desc=self.stage_name, file=sys.stdout, disable=not (self.verbose)) as iterator:
            for x, y in iterator:
                y = one_hot(y.long(), num_classes=4).permute(0, 3, 1, 2)
                y = y.float()
                x, y = x.to(self.device), y.to(self.device)
                loss, y_pred = self.batch_update(x, y)
```


Must add 

```
 y = one_hot(y.long(), num_classes=4).permute(0, 3, 1, 2)
                y = y.float()
```

oherwise ,the code will get wrong

**Comment by qubvel** [comment]
You can do onehot conversion in dataloader and not modify train.py

**Comment by qubvel** [comment]
activation should be `softmax2d` if your classes does not intersect and each pixel belong to some class

**Comment by HuangQinJian** [comment]
> activation should be `softmax2d` if your classes does not intersect and each pixel belong to some class

But in the readme ,you just mention that modify 

Change number of output classes in the model:

model = smp.Unet('resnet34', classes=3, activation='softmax')

**Comment by qubvel** [comment]
yes, it is most common case, but you also can use sigmoid or any other activation, it is up to you

**Comment by HuangQinJian** [comment]
> yes, it is most common case, but you also can use sigmoid or any other activation, it is up to you
I have a question,the sigmoid is just used to binary classify,the softmax is used as multify, I think it must use softmax

**Comment by qubvel** [comment]
You can use `sigmoid` to have multi label classification, e.g. if your image have `dog` and `cat` labels simultaneously. The same for segmentation if you pixel can belong to more than one class (e.g. `table` and `table leg`) use `sigmoid` activation. In this case you can imagine that your model have a lot of  heads with binary classificaton/segmentation problem each.

**Comment by HuangQinJian** [comment]
> You can use `sigmoid` to have multi label classification, e.g. if your image have `dog` and `cat` labels simultaneously. The same for segmentation, if you pixel can belong to more than one class (e.g. `table` and `table leg`) use `sigmoid` activation. In this case you can imagine that your model have a lot of heads with binary classificaton/segmentation problem each.

But my pixel can only belong to  one class

**Comment by qubvel** [comment]
So use softmax

**Comment by HuangQinJian** [comment]
> You can use `sigmoid` to have multi label classification, e.g. if your image have `dog` and `cat` labels simultaneously. The same for segmentation, if you pixel can belong to more than one class (e.g. `table` and `table leg`) use `sigmoid` activation. In this case you can imagine that your model have a lot of heads with binary classificaton/segmentation problem each.

> So use softmax

But when I used softmax,the result is so bad than using sigmoid,I do not why????

**Comment by qubvel** [comment]
Did  you include `background` class?

**Comment by gizemtanriver** [comment]
Hi @qubvel, my masks have 'lesion' labels and everything else should be treated as 'background'. Each pixel belongs to single class - either 'lesion' or 'background'. So in that case does it mean I have two classes and should use 'softmax' for activation? In the example notebook, why don't you include 'background' class?
